NuRadioReco/utilities/cr_flux.py

```diff
@@ -0,0 +1,261 @@
+    """
+
+    flux_interpolation = get_flux_interpolation(type=type)
+    return flux_interpolation(log10_energy)
+
+
+def get_analytic_cr_flux(log10_energy, type="auger_19"):
+    """
+    Parameters
+    ---------
+    log10_energy: float
+        energies (in log10(E / eV)))
+
+    type: string
+        choose between auger_17, auger_19 and TA_19
+    Returns
+    -------
+    analytic parametrization of spectrum in NuRadio base units (which are 1/(eV m^2 sr ns)
+    """
+    energy = 10 ** log10_energy
+
+    if type == "auger_17":
+        p = np.array([2.8e-19, 5.08e18, 39e18, 3.293, 2.53, 2.5])
+        spectrum = np.where(energy < p[1],
+                            p[0] * (energy / p[1]) ** (-p[3]),
+                            p[0] * (energy / p[1]) ** (-p[4]) * (1 + (p[1] / p[2]) ** p[5])
+                            * (1 + (energy / p[2]) ** p[5]) ** -1)
+    elif type == "auger_19":
+        p = np.array([3.46e12, 1.5e17, 6.2e18, 12e18, 50e18, 2.92, 3.27, 2.2, 3.2, 5.4])
+        spectrum = (energy / p[0]) ** (-p[5]) * \
+                   (1 + (energy / p[1]) ** p[5]) / (1 + (energy / p[1]) ** p[6]) * \
+                   (1 + (energy / p[2]) ** p[6]) / (1 + (energy / p[2]) ** p[7]) * \
+                   (1 + (energy / p[3]) ** p[7]) / (1 + (energy / p[3]) ** p[8]) * \
+                   (1 + (energy / p[4]) ** p[8]) / (1 + (energy / p[4]) ** p[9])
+    elif type == "TA_19":
+        p1 = -3.28
+        p2 = -2.68
+        p3 = -4.84
+        E1 = 10 ** 18.69
+        E2 = 10 ** 19.81
+        c = 2.24e-30
+        c1 = c * (E1 / 1e18) ** p1
+        c2 = c1 * (E2 / E1) ** p2
+        spectrum = np.where(energy < E1,
+                            c * (energy / 1e18) ** p1,
+                            np.where(energy < E2,
+                                     c1 * (energy / E1) ** p2,
+                                     c2 * (energy / E2) ** p3))
+
+    else:
+        raise NotImplementedError(f"Type {type} is not supported")
+
+    spectrum = spectrum * (units.eV * units.km ** 2 * units.sr * units.year) ** -1
+
+    return spectrum
+
+
+def get_flux_per_energy_bin(log10e_min, log10e_max, type='data_auger_19'):
+    """
+    Returns an scipy integration of the measured data or the analytic spectrum over given interval.
+
+    Parameters
+    ---------
+    log10e_min: float
+        Input energy lower bound (in log10(E / eV)))
+    log10e_max: float
+        Input energy upper bound (in log10(E / eV)))
+    type: string
+        choose between data_auger_19, data_TA_21, auger_17, auger_19, TA_19
+
+    Returns
+    -------
+    scipy integration of data in NuRadio base units (which are 1/(eV m^2 sr ns))
+    """
+    if type in ['data_auger_19', 'data_TA_21']:
+        E, J, J_err_low, J_err_up = get_measured_data(type)
+        flux_interpolation = get_flux_interpolation(type)
+        integrated_flux = quad(flux_interpolation, log10e_min, log10e_max,
+                               limit=2 * E.shape[0], points=E)
```

Comment:
... can you check if this is still correct (sorry for making you go from dE->dlogE in the interpolation, but:
dlogE/dE = 1/E, meaning if you change the integral from integrating over E to integrating over logE, you will have to likely multiply by E, because dE = E*dlogE, no?

Reply:
Nice catch! I will remove the option to get integration from data points since we have the analytic spectrum anyways.